src/main/java/io/debezium/connector/jdbc/JdbcSinkConnectorTask.java

```diff
@@ -122,6 +122,16 @@ public void put(Collection<SinkRecord> records) {
         LOGGER.trace("[PERF] Mark processed execution time {}", markProcessedStopWatch.durations());
     }
 
+    @Override
+    public void close(Collection<TopicPartition> partitions) {
+        for (TopicPartition partition : partitions) {
+            if (offsets.containsKey(partition)) {
+                LOGGER.trace("Requested close TopicPartition request for '{}'", partition);
+                offsets.remove(partition);
```

Comment:
Does it make sense to clear all the contents to be safe?

Reply:
Hi @blacktooth did you get an opportunity to give this patch a test-drive?

Reply:
@Naros  - I tested the fix. We see that topic partitions offsets  are cleared when connector tasks restarts. We are no more seeing 
`Ignoring invalid task provided offset topicXYZ-1 OffsetAndMetadata{offset=4966, leaderEpoch=null, metadata=''} -- partition not assigned, assignment=[topicXYZ-1] (org.apache.kafka.connect.runtime.WorkerSinkTask:434) ` exception in connector log @blacktooth

Reply:
Although it's being logged at `trace` level (in `markProcessed`), removing _all_ of the partitions would yield some potentially confusing/redundant messages if the task ends up being reused and one of its previous partition assignments was still present.

Reply:
Hi @aoswald you're referring to the `Requested close` followed by an `Advanced partition` message.  Would implementing the `open` method on the sink interface and logging the opening of the partition at trace level avoid that ambiguity you think?

Reply:
@Naros, I'm advocating your proposed approach of _not_ removing everything, but only partitions provided in the `close` method.

Reply:
In that case, I think we have a quorum on the path forward.  
Thank you both for the advocating & the testing.